bluemira/builders/EUDEMO/first_wall/divertor.py

```diff
@@ -0,0 +1,329 @@
+        # Just use the first scrape-off layer for now
+        point = find_point_along_wire_at_length(sols[0], leg_length)
+
+        # Create some vertical targets for now. Eventually the target
+        # angle will be derived from the grazing-angle parameter
+        target_length = self.params.div_Ltarg
+        target_coords = np.array(
+            [
+                [point[0], point[0]],
+                [point[1], point[1]],
+                [point[2] - target_length / 2, point[2] + target_length / 2],
+            ]
+        )
+        return PhysicalComponent(label, make_polygon(target_coords))
+
+    def make_dome(self, start: Sequence[float], end: Sequence[float], label: str):
+        """
+        Make a dome between the two given points
+        """
+        # Get the flux surface that crosses the through the start point
+        # We can use this surface to guide the shape of the dome
+        psi_start = self.equilibrium.psi(*start)
+        flux_surface = find_flux_surface_through_point(
+            self.equilibrium.x,
+            self.equilibrium.z,
+            self.equilibrium.psi(),
+            start[0],
+            start[1],
+            psi_start,
+        )
+
+        # Get the indices of the closest points on the flux surface to
+        # the input start and end points
+        start_coord = np.array([[start[0]], [start[1]]])  # [[x], [z]]
+        end_coord = np.array([[end[0]], [end[1]]])
+        idx = np.array(
+            [
+                np.argmin(np.hypot(*(flux_surface - start_coord))),
+                np.argmin(np.hypot(*(flux_surface - end_coord))),
+            ]
+        )
+
+        # Make sure the start and end are in the right order
+        if idx[0] > idx[1]:
+            idx = idx[::-1]
+            dome_contour = flux_surface[:, idx[0] + 1 : idx[1]]
+            dome_contour = dome_contour[:, ::-1]
+        else:
+            dome_contour = flux_surface[:, idx[0] + 1 : idx[1]]
+
+        # Build the coords of the dome in 3D (all(y == 0))
+        dome = np.zeros((3, dome_contour.shape[1] + 2))
+        dome[(0, 2), 0] = start_coord.T
+        dome[(0, 2), 1:-1] = dome_contour
+        dome[(0, 2), -1] = end_coord.T
+
+        return PhysicalComponent(label, make_polygon(dome))
+
+    def make_baffle(
+        self,
+        label: str,
+        start: Sequence[float],
+        end: Sequence[float],
+    ) -> PhysicalComponent:
+        """
+        Make a baffle.
+
+        Parameters
+        ----------
+        leg: LegPosition
+            The position of the leg the baffle should join to (e.g.,
+            inner).
+        start: Sequence
+            The position (in x-z) to start drawing the baffle from,
+            e.g., the outside end of a target.
+        end: Sequence
+            The position (in x-z) to stop drawing the baffle, e.g., the
+            postion to the upper part of the first wall.
+        """
+        # Just generate the most basic baffle: a straight line
```

Comment:
Perhaps this comment can go into the docstring for the function? Then the docs will also reflect that this is currently making a baffle using straight lines?